test/assets.js

```diff
@@ -0,0 +1,523 @@
+      scriptAsm: '30440220761464d7bab9515d92260762a97af82a9b25d202d8f7197b1aaec81b6fed541f022059f99606de6b06e17b2cd102dceb3807ebdd9e777a5b77c9a0b3672f5eabcb3101 04eb1e0ccd9afcac42229348dd776e991c69551ae3474340fada12e787e51758397e1d3afdba360d6374261125ea3b6ea079a5f202c150dfd729e1062d9176a307',
+      satoshis: 16419885,
+    },
+  ],
+  outputs: [
+    {
+      satoshis: 21247964,
+      script: '76a9144b7b335f978f130269fe661423258ae9642df8a188ac',
+      scriptAsm: 'OP_DUP OP_HASH160 4b7b335f978f130269fe661423258ae9642df8a1 OP_EQUALVERIFY OP_CHECKSIG',
+      address: 'mnQ4ZaGessNgdxmWPxbTHcfx4b8R6eUr1X',
+    },
+    {
+      address: 'mqdofsXHpePPGBFXuwwypAqCcXi48Xhb2f',
+      satoshis: 13677426,
+      scriptAsm: 'OP_DUP OP_HASH160 6efcf883b4b6f9997be9a0600f6c095fe2bd2d92 OP_EQUALVERIFY OP_CHECKSIG',
+      script: '76a9146efcf883b4b6f9997be9a0600f6c095fe2bd2d9288ac',
+      spentTxId: spentTxId,
+      spentIndex: spentIndex,
+      spentHeight: 10,
+    },
+  ],
+  locktime: 0,
+};
+
+const utxos = [
+  {
+    "addresses": [
+      "mwrPT9XidZ4KvtdfvbkCaSS5GtakZzXp6J",
+    ],
+    "txid": "a3003756db8dd832087b98c5adcce8e1fade89959519713edecccadc44a4d2f4",
+    "value": 30000,
+  },
+  {
+    "addresses": [
+      "mwrPT9XidZ4KvtdfvbkCaSS5GtakZzXp6J",
+    ],
+    "txid": "a3003756db8dd832087b98c5adcce8e1fade89959519713edecccadc44a4d2f4",
+    "value": 25000,
+  },
+  {
+    "addresses": [
+      "mwrPT9XidZ4KvtdfvbkCaSS5GtakZzXp6J",
+    ],
+    "txid": "32e3c6c28b50183c9b407fb3ab4d83ea0d3ad195325253784772beea0db82c34",
+    "value": 20000,
+  },
+  {
+    "addresses": [
+      "mwrPT9XidZ4KvtdfvbkCaSS5GtakZzXp6J",
+    ],
+    "txid": "a3003756db8dd832087b98c5adcce8e1fade89959519713edecccadc44a4d2f4",
+    "value": 15000,
+  },
+  {
+    "addresses": [
+      "mwrPT9XidZ4KvtdfvbkCaSS5GtakZzXp6J",
+    ],
+    "txid": "a3003756db8dd832087b98c5adcce8e1fade89959519713edecccadc44a4d2f4",
+    "value": 15000,
+  },
+];
+
+describe.only('AssetController', () => {
+  describe('_calculateMiningFee', () => {
+    const node = {};
+    const controller = new AssetController(node);
+    let ins = 1;
+    let outs = 30;
+    let feeRate = 100000;
+    let fee;
+
+    it('should calculate and return the Mining Fee', () => {
+      let miningFee = controller._calculateMiningFee(ins, outs, feeRate);
+      fee = miningFee;
+      should(fee).be.a.Number();
+      miningFee = controller._calculateMiningFee(++ins, outs, feeRate);
+      should(miningFee).be.exactly(fee + 18000);
+      fee = miningFee;
+      miningFee = controller._calculateMiningFee(++ins, ++outs, feeRate);
+      should(miningFee).be.exactly(fee + 18000 + 3400);
+    });
+  });
+
+  describe('_calculateInputsValue', () => {
+    it('should return the total sum of utxos value', () => {
+      const inputsValue = AssetController.prototype._calculateInputsValue(utxos);
+      should(inputsValue).be.exactly(105000);
+    });
+  });
+
+  describe('_evaluateInputsValue', () => {
```

Comment:
It makes use of the actual implementations _calculateMiningFee and _calculateInputValue, testing more than one implementation, it would be better to use a mock